packages/as-types/assembly/amount.ts

```diff
@@ -28,62 +32,104 @@ export class Amount {
   /**
    * Adds two amounts and return results in a new one.
    *
-   * @param  a - Amount to add.
+   * @param  other - The other Amount to add.
+   *
+   * @see {@link Result} for more information about the return type.
+   *
+   * @returns
+   * A Result object containing:
+   * - The new Amount if operation was successful.
+   * - An error message if:
+   *    - the Amount objects are not the same.
+   *    - the addition would overflow of the 'u64.MAX_VALUE' limit.
+   *
    */
-  add(a: Amount): Result<Amount> {
-    if (this.currency != a.currency) {
+  @operator('+')
+  add(other: Amount): Result<Amount> {
+    if (this.currency != other.currency) {
       return new Result(
         new Amount(),
         'the sum is impossible: the amounts have different currencies',
       );
     }
 
-    if (a.value > u64.MAX_VALUE - this.value) {
+    if (other.value > u64.MAX_VALUE - this.value) {
       // tests overflow
       return new Result(new Amount(), 'the sum is impossible: overflow');
     }
 
-    return new Result(new Amount(this.value + a.value, this.currency));
+    return new Result(new Amount(this.value + other.value, this.currency));
   }
 
   /**
-   * Substracts two amounts and return results in a new one.
+   * Subtracts two amounts and return results in a new one.
+   *
+   * @see {@link Result} for more information about the return type.
    *
-   * @param  a - Amount to substract.
+   * @param  other - Amount to subtract.
+   *
+   * @returns
+   * A Result object containing:
+   * - The new Amount if the operation was successful.
+   * - An error message if:
+   *    - the currency is not the same between the two amounts.
+   *    - the subtraction would underflow .
    */
-  substract(a: Amount): Result<Amount> {
-    if (this.currency != a.currency) {
+  @operator('-')
+  subtract(other: Amount): Result<Amount> {
```

Comment:
There was a typo in the function name is it ok to change it ?
(added corresponding operator also for both add and subtract functions)
@gregLibert @Ben-Rey

Reply:
It should be corrected as you did